js/components/ImportIndicatorsPopover.js

```diff
@@ -1,142 +1,271 @@
-import React from 'react';
+import React, { useState, useContext, useEffect } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { BootstrapPopoverButton } from './helpPopover';
 import api from '../apiv2';
 
-class ImportIndicatorsPopover extends React.Component {
-    // These define the different cases/views of the Popover to switch between.
-    DOWNLOAD_UPLOAD = 0;
-    FEEDBACK = 1;
-    CONFIRM = 2;
-    SUCCESS = 3;
+export const ImportIndicatorsContext = React.createContext();
+
+export class ImportIndicatorsButton extends BootstrapPopoverButton {
+    popoverName = "importIndicators"
+    popoverTitle = "Import indicators"
 
     constructor(props) {
         super(props);
         this.state = {
-            status: this.DOWNLOAD_UPLOAD
+            levelStore: this.props.levelStore, 
+            chosenTier: this.props.levelStore.tierTemplates[this.props.levelStore.chosenTierSetKey].tiers, // State containing a list of tier levels names of the chosen RF framework tier template
+            tierLevelsRows: [], // State to hold the tier levels name and the desired number of rows for the excel template
+            setTierLevelsRows: this.setTierLevelsRows, // Method used to update the number of row per tier level
         }
     }
 
-    handleDownload = () => {
+    setTierLevelsRows = (i, tier, value) => {
+        let updatedTiers = this.state.tierLevelsRows;
+        updatedTiers[i] = {[tier] : value};
+        this.setState({ tierLevelsRows: updatedTiers });
+    }
+
+    getPopoverContent = () => {
```

Comment:
Maybe just add a comment that this is overriding a method in the BootstrapPopoverButton.